robot_calibration/src/capture_features.cpp

```diff
@@ -0,0 +1,304 @@
+// Author: Jack Center
+
+#include <fstream>
+#include <ctime>
+
+#include <ros/ros.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <rosbag/query.h>
+
+#include <std_msgs/String.h>
+#include <robot_calibration_msgs/CalibrationData.h>
+#include <robot_calibration_msgs/CaptureConfig.h>
+
+#include <robot_calibration/capture/chain_manager.h>
+#include <robot_calibration/capture/feature_finder_loader.h>
+
+#include <camera_calibration_parsers/parse.h>
+#include <robot_calibration/ceres/optimizer.h>
+#include <robot_calibration/camera_info.h>
+#include <robot_calibration/load_bag.h>
+
+#include <boost/foreach.hpp>  // for rosbag iterator
+
+#include <robot_calibration/capture_features.h>
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv,"robot_calibration");
+  ros::NodeHandle nh("~");
+
+  if (!check_parameters(nh))
+    return -1;
```

Comment:
Two small things,  when capture features falis here before you return -1 you should probably log an error.  Second is that even in the case where the if statement only has one line following it, it is good practice to add brackets around it `{}` because then when you add a second line (like I'm asking with the logging statement) it will work as you intend.